src/utils/ssr/getAuthenticationInfo.js

```diff
@@ -0,0 +1,98 @@
+import {
+  CognitoIdentityClient,
+  ListIdentityPoolsCommand,
+} from '@aws-sdk/client-cognito-identity';
+
+import {
+  CognitoIdentityProviderClient,
+  ListUserPoolsCommand,
+  ListUserPoolClientsCommand,
+  DescribeUserPoolClientCommand,
+  DescribeUserPoolCommand,
+} from '@aws-sdk/client-cognito-identity-provider';
+
+import { getDefaultRoleAssumerWithWebIdentity } from '@aws-sdk/client-sts';
+import { fromTokenFile } from '@aws-sdk/credential-provider-web-identity';
+import loadAuthenticationInfo from '../../redux/actions/networkResources/loadAuthenticationInfo';
+
+const getAuthenticationInfo = async (context, store) => {
+  if (
+    store.getState().networkResources.auth.userPoolId
+    || store.getState().networkResources.auth.identityPoolId
+  ) {
+    return;
+  }
+
+  let additionalClientParams = {};
+
+  if (process.env.NODE_ENV !== 'development') {
+    additionalClientParams = {
+      ...additionalClientParams,
+      credentials: fromTokenFile({
+        roleAssumerWithWebIdentity: getDefaultRoleAssumerWithWebIdentity(),
+      }),
+    };
+  }
+
+  const identityPoolClient = new CognitoIdentityClient(
+    {
+      region: 'eu-west-1',
```

Comment:
this region `eu-west-1` looks hardcoded in multiple places atm, should we move it to a config file? There are a couple of other things like `sandboxId` on line 60 as well. I wonder though of you are more eager to get this PR through and do this as part of a separate work?

Reply:
I just saw you actually have `src/utils/amplify-config.js`, maybe you can get that from there?

Reply:
This needs to be a separate task per our discussion. Resolving accordingly.